Add attachment data with content ids

mlabarca opened the pull request:
The content ids is valuable information that we should be able to reliable access from the parsed email. this pr adds an array of hashes with the content id, type and file instead of just an array of files, while keeping the attachment array we already use in griddler.  This can be useful when using a separate email class with this data. 

Todo: I still need to figure out how to make this work the default email class in griddler, just opening this pr to check how feasible is this.

mlabarca committed:
Add attachment data with content ids

mlabarca commented:
I'm thinking to change this to save this data in [this](https://github.com/thoughtbot/griddler/blob/7690cc31ded0f834b77160f1d85217b85d3480cd/lib/griddler/email.rb#L21)  vendor specific param on the griddler email class.

sdhull commented:
This seems reasonable to me.

Also it seems reasonable to assign it to `vendor_specific: {attachment_info: {info...}}` (I'd name it `attachment_info` rather than `attachment_data` to match what ppl see in sendgrid docs).

And would be reasonable to add some assertions about the extracted info in specs (beyond just `.present?`)

mlabarca commented:
Sounds good, I'll get back to this tomorrow and I'll add some more
assertions.

On Thu, May 23, 2019, 9:49 PM Steve <notifications@github.com> wrote:

> This seems reasonable to me.
>
> Also it seems reasonable to assign it to vendor_specific:
> {attachment_info: {info...}} (I'd name it attachment_info rather than
> attachment_data to match what ppl see in sendgrid docs).
>
> And would be reasonable to add some assertions about the extracted info in
> specs (beyond just .present?)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/thoughtbot/griddler-sendgrid/pull/35?email_source=notifications&email_token=ABDADPKRFALVNHTOFW4AZ2LPW5CUHA5CNFSM4HPK5AR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWD56SI#issuecomment-495443785>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABDADPJYJ67D4OVJC6TVPOLPW5CUHANCNFSM4HPK5ARQ>
> .
>

mlabarca committed:
Move attachment info into vendor-specific

mlabarca committed:
Improve tests with new vendor-specific attributes

mlabarca commented:
There, I changed this to add a new array `attachment_info`  under vendor_specific with the same `file` that we are sending on  the `attachments`, but now we have its precise `content_id` and `type`. I also added a couple of methods to the specs since it'd be the third time we use that large `eojson` string with the attachment-info.

mlabarca commented:
Note the change that it doesn't do a `params.delete` the attachment1, ...attachmentn params anymore. Was that necessary for griddler somehow?

stefannibrasil commented:
Closing this as this library has been deprecated: https://github.com/thoughtbot/griddler-sendgrid/pull/41
